### PR TITLE
Add mlpack

### PR DESCRIPTION
**Pull request by marcelotrevisani**

### CODE REVIEW

**Comment by marcelotrevisani** [comment]
@conda-forge/staged-recipes please, review

**Comment by scopatz** [comment]
LGTM, don't worry about the linter failure!

File: recipes/mlpack/meta.yaml [hunk 1]
```diff
@@ -20,20 +20,20 @@ requirements:
...
+    - armadillo >=6.500.0
     - pytest-runner
   run:
     - boost
     - python
     - numpy
-    - pandas
-    - armadillo
+    - pandas  >=0.15.0
+    - armadillo >=6.500.0
```
Comment (isuruf) [inline recipes/mlpack/meta.yaml]: This should be `{{ pin_compatible('armadillo') }}`. Can be done in the feedstock.

Comment (marcelotrevisani) [reply recipes/mlpack/meta.yaml]: got it! Thanks

**Pull request merged**